main.go

```diff
@@ -48,143 +49,132 @@ Usage:
 
 	flag.Parse()
 
-	queue := make(chan string)
-
 	if *hostOnly {
 		fmt.Printf("Crawling URL[%s]\n", *link)
 	} else {
 		fmt.Printf("Crawling All Links in URL[%s]\n", *link)
 	}
 
-	// Go routine puts url flag into channel for queuing.
-	go func() {
-		queue <- *link
-	}()
-
-	hostURL, err := url.Parse(*link)
+	path, err := url.Parse(*link)
 	if err != nil {
+		fmt.Printf("Error: Unparsable URI: %s\n", *link)
+		os.Exit(-1)
 		return
 	}
 
+	// deadLinks provides a lists to contain all failed url links that are
+	// considered deadlinks.
+	var deadLinks []*linkReport
+
+	dead := collectFrom(path, !*hostOnly)
```

Comment:
The function collectFrom takes the path and pulls out the links for that page, using that first set it recursively crawls their sublinks until the master link list is done and it closes the channel, even if all were possible it is guaranteed to end once done